libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py

```diff
@@ -20,40 +20,46 @@
 from urllib.parse import urlparse, urlunparse
 
 import requests
-from langchain_core.pydantic_v1 import (
+from pydantic import (
     BaseModel,
+    ConfigDict,
     Field,
     PrivateAttr,
     SecretStr,
-    validator,
+    field_validator,
 )
 from requests.models import Response
 
 from langchain_nvidia_ai_endpoints._statics import MODEL_TABLE, Model, determine_model
 
 logger = logging.getLogger(__name__)
 
+_API_KEY_VAR = "NVIDIA_API_KEY"
+_BASE_URL_VAR = "NVIDIA_BASE_URL"
+
 
 class _NVIDIAClient(BaseModel):
     """
     Low level client library interface to NIM endpoints.
     """
 
     default_hosted_model_name: str = Field(..., description="Default model name to use")
-    model_name: Optional[str] = Field(..., description="Name of the model to invoke")
+    # "mdl_name" because "model_" is a protected namespace in pydantic
+    mdl_name: Optional[str] = Field(..., description="Name of the model to invoke")
     model: Optional[Model] = Field(None, description="The model to invoke")
     is_hosted: bool = Field(True)
     cls: str = Field(..., description="Class Name")
 
     # todo: add a validator for requests.Response (last_response attribute) and
     #       remove arbitrary_types_allowed=True
-    class Config:
-        arbitrary_types_allowed = True
+    model_config = ConfigDict(
+        arbitrary_types_allowed=True,
+    )
```

Comment:
We can also add `protected_namespaces=()` to remove warnings if we want to keep existing field names
```suggestion
    model_config = ConfigDict(
        arbitrary_types_allowed=True,
        protected_namespaces=(),
    )
```
We did this on all instances of [RunnableSerializable](https://github.com/langchain-ai/langchain/blob/a47b33284152d85d2233f069f993f9c0ef4da0d7/libs/core/langchain_core/runnables/base.py#L2456-L2460) in langchain-core.

Reply:
i considered that and figured this since is internal code (not user visible), it was fine to leave the namespace protection in place.

unfortunately, i didn't see a way to exclude a specific field, which would be the best solution imho.